src/main/java/com/github/javalab/javaiaas/controllers/ApplicationsController.java

```diff
@@ -0,0 +1,74 @@
+package com.github.javalab.javaiaas.controllers;
+
+
+import com.github.javalab.javaiaas.models.Application;
+import com.github.javalab.javaiaas.models.User;
+import com.github.javalab.javaiaas.security.details.UserDetailsImpl;
+import com.github.javalab.javaiaas.services.ApplicationService;
+import javassist.NotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/applications")
+public class ApplicationsController {
+    private final ApplicationService service;
+
+    public ApplicationsController(ApplicationService service) {
+        this.service = service;
+    }
+
+    @GetMapping
+    public ResponseEntity<?> index(Authentication authentication) {
+        return ResponseEntity.ok(currentUser(authentication).getApplications());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<?> show(Authentication authentication,
+                                  @PathVariable Long id) throws NotFoundException {
+        authorize(authentication, id);
```

Comment:
maybe change on @PreAuthorize("isAuthenticated()") ?

Reply:
But how to check if current user owns the application?